### PR TITLE
Handle differences between JEM and JES models.

### PR DESCRIPTION
**Pull request by kadams54**
JEM and JES models have different data schemas. Deal with it.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4750/
Test PASSed.

**Comment by kadams54** [comment]
## QA

You'll need to connect to JEM in staging. If you haven't already, edit your development.ini to change the `jujugui.jem_url` setting to the one pointing to staging.

1. Head to `/profile/:flags:/blues/`.
2. You should see at least two models listed: `jcsackett/amazon` and `yellow/aws-eu-central-1`. Click on one to manage and verify that the dropdown has its values populated correctly.
3. Click on the "Manage" button and verify that you successfully switch to the environment in question.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -1872,9 +1872,6 @@ YUI.add('juju-gui', function(Y) {
     */
     switchEnv: function(socketUrl, username, password) {
       console.log('switching to new socket URL:', socketUrl);
-      if (this.get('sandbox')) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Why was this removed? I'm pretty sure it's still not supported.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 2]
```diff
@@ -258,7 +270,7 @@ YUI.add('user-profile', function() {
       return (
         <juju.components.UserProfileEntity
           entity={model}
-          expanded={model.name === this.props.currentModel}
+          expanded={name === this.props.currentModel}
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: `name` isn't defined.

Comment (kadams54) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Good catch.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4751/
Test PASSed.

File: jujugui/static/gui/src/app/app.js (hunk 1 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/app.js]: It is supported, if you point your sandbox at a real JEM (as specified by the QA instructions).

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 3]
```diff
@@ -140,7 +140,19 @@ YUI.add('user-profile', function() {
...
-      this.setState({envList: data.envs || data});
+      var envs = data.envs || data;
+      // XXX kadams54: JEM models don't *currently* have a name or owner. They
+      // have a path which is a combination of both, but that format may change
+      // on down the road. Hence this big comment.
+      var envList = envs.map((env) => {
+        if (env.path) {
+          env.name = env.path;
+          env.owner = env.path.split('/')[0];
+          env.lastConnection = 'NA';
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: N/A :)

File: jujugui/static/gui/src/app/components/user-profile/test-user-profile.js [hunk 4]
```diff
@@ -494,7 +501,20 @@ describe('UserProfile', () => {
...
+    var output = component.getRenderOutput();
+    var displayedModel = output.props.children.props.children.props.children[1].props.children[0].props.children[1].props.children[1][0]; // eslint-disable-line
+    var displayedChildren = displayedModel.props.children;
+    var displayedName = displayedChildren[0].props.children;
+    var displayedLastConnection = displayedChildren[2].props.children;
+    var displayedOwner = displayedChildren[4].props.children;
+    var jemModel = jemModels[0];
+    assert.equal(displayedName, jemModel.path);
+    // XXX kadams54: Because we don't have a LastConnection yet for JEM models.
+    assert.equal(displayedLastConnection, 'NA');
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/user-profile/test-user-profile.js]: N/A

File: jujugui/static/gui/src/app/components/user-profile/test-user-profile.js [hunk 5]
```diff
@@ -494,7 +501,20 @@ describe('UserProfile', () => {
     var instance = component.getMountedInstance();
 
     assert.equal(jem.listEnvironments.callCount, 1);
-    assert.deepEqual(instance.state.envList, models);
+    assert.deepEqual(instance.state.envList, jemModels);
+
+    // Make sure we properly displayed the different bits within a JEM model.
+    var output = component.getRenderOutput();
+    var displayedModel = output.props.children.props.children.props.children[1].props.children[0].props.children[1].props.children[1][0]; // eslint-disable-line
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/test-user-profile.js]: gross - maybe we should investigate some of those libraries for querying react components for situations like this.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4752/
Test PASSed.

File: jujugui/static/gui/src/app/components/user-profile/test-user-profile.js (hunk 5 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/components/user-profile/test-user-profile.js]: +1,000,000,000

**Comment by hatched** [comment]
:+1: :ok:
Thanks for this fix!

**Comment by urosj** [comment]
:+1:

File: jujugui/static/gui/src/app/components/user-profile/test-user-profile.js (hunk 5 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/components/user-profile/test-user-profile.js]: Like https://github.com/jquense/teaspoon.
